### PR TITLE
ci(workflow): add cache to workflows using actions/setup-node

### PR DESCRIPTION
**Pull request by oscard0m**
## Description

Add `cache` to workflows using `actions/setup-node`

## Context

`setup-node` GitHub Action just released a new option to add cache to steps using it.

You can find the details here: https://github.blog/changelog/2021-07-02-github-actions-setup-node-now-supports-dependency-caching/

---

🤖 This PR has been generated automatically by [this octoherd script](https://github.com/oscard0m/octoherd-script-add-cache-to-node-github-action), feel free to run it in your GitHub user/org repositories! 💪🏾

### CODE REVIEW

**Review by dj110919871 — APPROVED** [review]

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -3,14 +3,14 @@ name: build-test
...
-  push:    
+      - "**.md"
+  push:
     branches:
       - main
       - releases/*
     paths-ignore:
-      - '**.md'
-      
+      - "**.md"
```
Comment (dmitry-shibanov) [inline .github/workflows/build-test.yml:12]: Hello @oscard0m . Thank you for your pull request. Could you please revert this changes ?

**Review by MaksimZhukov — APPROVED** [review]
